cinch/playbooks/jenkins_restart_utility.yml

```diff
@@ -0,0 +1,139 @@
+# This playbook aims to gather diagnostics data in the event of
+# trouble with Jenkins, and then restarts Jenkins to keep operations running.
+# By default, only the diagnostics data is collected, but the Jenkins service
+# restart can be enabled by setting the 'restart_jenkins' variable to true:
+
+# Example:
+# ansible-playbook -u root --private-key /path/to/key
+# -i '10.0.183.169', jenkins_restart_utility.yml -e "restart_jenkins=true"
+
+- hosts: all
+  vars:
+    jenkins_pid_path: '/run/jenkins.pid'
+  tasks:
+    # This block ignores all errors to ensure that Jenkins is restarted on
+    # demand even if we can't gather diagnostics data.
+    - name: gather diagnostics data
+      block:
+        - name: check if Jenkins is running
+          stat:
```

Comment:
If stating the PID file isn't a guarantee that the process is up, perhaps we could change the value of the register on this task to make it clear that this is only testing for the existence of the file. So naming it something like ``jenkins_pid_file`` instead of ``jenkins_running`` could make that more clear.